### PR TITLE
[Merged by Bors] - feat: Add polar change of coordinates for complex variable

### PR DESCRIPTION
**Pull request by xroblot**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Measure/Lebesgue/Complex.lean [hunk 1]
```diff
@@ -84,4 +83,45 @@ theorem volume_closedBall (a : ℂ) (r : ℝ) :
...
+    Complex.equivRealProdAddHom.symm p = p.1 + p.2 * Complex.I := Complex.equivRealProd_symm_apply p
+
+theorem equivRealProdLm_symm_apply (p : ℝ × ℝ) :
+    Complex.equivRealProdLm.symm p = p.1 + p.2 * Complex.I := Complex.equivRealProd_symm_apply p
+
+theorem equivRealProdClm_symm_apply (p : ℝ × ℝ) :
+    Complex.equivRealProdClm.symm p = p.1 + p.2 * Complex.I := Complex.equivRealProd_symm_apply p
+
+protected theorem polarCoord_source :
+    Complex.polarCoord.source = {a | 0 < a.re} ∪ {a | a.im ≠ 0} := by simp [Complex.polarCoord]
```
[the remark this answers is not in the document]
Comment (xroblot) [reply Mathlib/MeasureTheory/Measure/Lebesgue/Complex.lean:99]: Done.

File: Mathlib/MeasureTheory/Measure/Lebesgue/Complex.lean [hunk 2]
```diff
@@ -84,4 +83,45 @@ theorem volume_closedBall (a : ℂ) (r : ℝ) :
...
+  rfl
+
+theorem equivRealProdAddHom_symm_apply (p : ℝ × ℝ) :
+    Complex.equivRealProdAddHom.symm p = p.1 + p.2 * Complex.I := Complex.equivRealProd_symm_apply p
+
+theorem equivRealProdLm_symm_apply (p : ℝ × ℝ) :
+    Complex.equivRealProdLm.symm p = p.1 + p.2 * Complex.I := Complex.equivRealProd_symm_apply p
+
+theorem equivRealProdClm_symm_apply (p : ℝ × ℝ) :
+    Complex.equivRealProdClm.symm p = p.1 + p.2 * Complex.I := Complex.equivRealProd_symm_apply p
```
[the remark this answers is not in the document]
Comment (xroblot) [reply Mathlib/MeasureTheory/Measure/Lebesgue/Complex.lean]: I moved these just after the definition of the correspond functions.

File: Mathlib/MeasureTheory/Measure/Lebesgue/Complex.lean [hunk 3]
```diff
@@ -84,4 +83,38 @@ theorem volume_closedBall (a : ℂ) (r : ℝ) :
...
+
+protected theorem polarCoord_apply (a : ℂ) :
+    Complex.polarCoord a = (Complex.abs a, Complex.arg a) := by
+  simp_rw [Complex.abs_def, Complex.normSq_apply, ← pow_two]
+  rfl
+
+protected theorem polarCoord_source :
+    Complex.polarCoord.source = {a | 0 < a.re} ∪ {a | a.im ≠ 0} := by simp [Complex.polarCoord]
+
+alias polarCoord_target := polarCoord_target
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Measure/Lebesgue/Complex.lean:101]: I'm sure this line does what I hope it does, but that's still a lot of obfuscation, and also the name is probably not protected which could lead to clashes. Maybe better to have a true theorem
```
protected theorem polarCoord_target :
    Complex.polarCoord.target = ... := by
  ...
```
?

File: Mathlib/MeasureTheory/Measure/Lebesgue/Complex.lean [hunk 4]
```diff
@@ -3,9 +3,8 @@ Copyright (c) 2021 Yury Kudryashov. All rights reserved.
 Released under Apache 2.0 license as described in the file LICENSE.
 Authors: Yury Kudryashov
 -/
-import Mathlib.Analysis.SpecialFunctions.Integrals
+import Mathlib.Analysis.SpecialFunctions.PolarCoord
```
Comment (sgouezel) [inline Mathlib/MeasureTheory/Measure/Lebesgue/Complex.lean:6]: When you import this, you import a lot of nontrivial things. Could you instead leave the file as it was, and put your results in the file `Mathlib.Analysis.SpecialFunctions.PolarCoord`?

File: Mathlib/MeasureTheory/Measure/Lebesgue/Complex.lean (hunk 3 above)
Comment (xroblot) [reply Mathlib/MeasureTheory/Measure/Lebesgue/Complex.lean]: Indeed, it is a bit confusing. I fixed that.

File: Mathlib/MeasureTheory/Measure/Lebesgue/Complex.lean (hunk 4 above)
Comment (xroblot) [reply Mathlib/MeasureTheory/Measure/Lebesgue/Complex.lean]: I moved everything to `Mathlib.Analysis.SpecialFunctions.PolarCoord` as you suggested. That means of course that I need to import `Mathlib.MeasureTheory.Measure.Lebesgue.Complex` in this file but I guess that's okay.

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**